folium/features.py

```diff
@@ -659,6 +688,14 @@ def __init__(
                 raise TypeError(
                     "Only Marker, Circle, and CircleMarker are supported as GeoJson marker types."
                 )
+
+        if popup_keep_highlighted:
+            if popup is None:
+                raise NameError(
```

Comment:
not sure if `NameError` is the right class here. Docs say it's for missing variables, which doesn't apply here: the variable exists, it just doesn't have the right value. I'd say go with a `ValueError` instead.